packages/components/inputs/select-utils/src/types.ts

```diff
@@ -0,0 +1,15 @@
+import type SelectComponents from 'react-select';
+
+type ComponentProps<T> = T extends React.ComponentType<infer P> ? P : never;
+
+export type TReactSelectCustomComponentsProps = {
+  [K in keyof SelectComponents]: ComponentProps<SelectComponents[K]>;
+}[keyof SelectComponents];
+
+export type TSelectInputCustomComponentProps<
```

Comment:
This is a generic type that extends TReactSelectCustomComponentsProps. It represents the props for a custom component that can be used as an input in a select component. It adds a selectProps property to the props type T, which includes all the properties from T['selectProps'] and an optional isCondensed property. This allows you to pass additional props to the select component through the selectProps property.